scripts/core/editor3/components/tests/editor3.spec.jsx

```diff
@@ -33,6 +33,51 @@ describe('editor3.component', () => {
             expect(onDragOver(makeEvent(invalidType))).toBe(true);
         });
     });
+
+    it('should not accept dragging when editor is readOnly', () => {
+        const wrapper = shallow(<Editor3 readOnly={true} editorFormat={['picture']} singleLine={false}/>);
+        const {onDragOver} = wrapper.instance();
+        const makeEvent = (t) => ({originalEvent: {dataTransfer: {types: [t]}}});
+
+        [
+            'application/superdesk.item.picture',
+            'application/superdesk.item.graphic',
+            'application/superdesk.item.video',
+            'text/html'
+        ].forEach((validType) => {
+            expect(onDragOver(makeEvent(validType))).toBe(true);
+        });
+    });
+
+    it('should not accept dragging when editor does not support images', () => {
+        const wrapper = shallow(<Editor3 readOnly={false} editorFormat={[]} singleLine={false}/>);
+        const {onDragOver} = wrapper.instance();
+        const makeEvent = (t) => ({originalEvent: {dataTransfer: {types: [t]}}});
+
+        [
+            'application/superdesk.item.picture',
+            'application/superdesk.item.graphic',
+            'application/superdesk.item.video',
+            'text/html'
+        ].forEach((validType) => {
+            expect(onDragOver(makeEvent(validType))).toBe(true);
+        });
+    });
+
+    it('should not accept dragging when editor is single line', () => {
+        const wrapper = shallow(<Editor3 readOnly={false} editorFormat={['picture']} singleLine={true}/>);
+        const {onDragOver} = wrapper.instance();
+        const makeEvent = (t) => ({originalEvent: {dataTransfer: {types: [t]}}});
+
+        [
+            'application/superdesk.item.picture',
+            'application/superdesk.item.graphic',
+            'application/superdesk.item.video',
+            'text/html'
+        ].forEach((validType) => {
+            expect(onDragOver(makeEvent(validType))).toBe(true);
```

Comment:
Same for this test. You can also use `.toBeTruthy()` and `.toBeFalsy()` instead of `.toBe(true)` and `.toBe(false)` if you want.